src/app/components/itinerary/itinerarylist/itinerarylist.component.ts

```diff
@@ -11,37 +11,47 @@ import { TokenService } from 'src/app/services/token.service';
 })
 export class ItinerarylistComponent implements OnInit {
 
-  constructor(private itineraryService: ItineraryService,private route: ActivatedRoute, private tokenService: TokenService) { }
+  constructor(private itineraryService: ItineraryService, private activatedRoute: ActivatedRoute, private tokenService: TokenService, private router: Router) { }
 
   itineraries: Itinerary[] = [];
   numberOfElements: Number;
-  totalPages: Array<Number>=[];
+  totalPages: Array<Number> = [];
   pageId: Number;
+  currentUrl: String;
 
   @Input()
-  userId: Number;
-  
   username: String;
 
+  @Input()
+  base_url: String;
 
-  ngOnInit(): void {
-    this.pageId = Number(this.route.snapshot.paramMap.get('id')) -1;
-    this.username=this.tokenService.getUsername()
 
-    this.loadUserItineraries(this.pageId,this.userId);
+  ngOnInit(): void {
+    this.pageId = Number(this.activatedRoute.snapshot.paramMap.get('id')) - 1;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ispp-yourney/angular-app/pullRequest?prid=7181827)
```suggestion
    this.pageId = Number(this.activatedRoute.snapshot.paramMap.get("id")) - 1;
```